src/Microsoft.Framework.DependencyInjection.Abstractions/ServiceProviderExtensions.cs

```diff
@@ -82,7 +82,14 @@ public static IEnumerable<T> GetRequiredServices<T>([NotNull] this IServiceProvi
         public static IEnumerable<object> GetRequiredServices([NotNull] this IServiceProvider provider, [NotNull] Type serviceType)
         {
             var genericEnumerable = typeof(IEnumerable<>).MakeGenericType(new [] { serviceType });
-            return (IEnumerable<object>)provider.GetRequiredService(genericEnumerable);
+            var providers = (IEnumerable<object>)provider.GetRequiredService(genericEnumerable);
+            
+            if (!providers.Any())
```

Comment:
This just in, from the world of premature optimization...

What perf risks are there from calling `.Any()` on this? I have this fear that common implementations of `IServiceProvider` might do costly work every time the result is enumerated. And with this change, the work might be done twice. I'm figuring this method here is called a zillion times per request, so that's why I have this concern.